### ISSUE TITLE
Passing instance of pg.ClientBase to transaction functions results in runtime error

### ISSUE THREAD

**Issue by tetchel** [jawj/zapatos#107]
Hi, thanks for the great library

I think I have encountered a bug with the type signature of the transaction functions, in my case I'm using `db.serializable`

I had been passing a `pg.PoolClient` instance around to my zapatos calls ie 

```ts
db.select('customer', db.all).run(ctx.client)
```

When I try do this for a transaction:
```ts
   const result = await db.serializable(ctx.client, async (txnClient) => {

      const newContact = await db.insert('contact', {
        // ... contact object etc
      }).run(txnClient);
```

it fails at runtime with the following error:
```
      "message": "Client has already been connected. You cannot reuse a client.",
      "stack":
          Error: Client has already been connected. You cannot reuse a client.
              at Client._connect (<project>/backend/node_modules/pg/lib/client.js:94:19)
              at <project>/backend/node_modules/pg/lib/client.js:166:12
              at new Promise (<anonymous>)
              at Client.connect (<project>/backend/node_modules/pg/lib/client.js:165:12)
              at transaction (<project>/backend/node_modules/zapatos/dist/db/transaction.js:44:63)
              at Object.serializable (<project>/backend/node_modules/zapatos/dist/db/transaction.js:95:12)
```

It works when I use a pool instead of a client.

I see all the examples in the documentation pass a `pg.Pool`, so it may be expected that the transaction functions require a connection pool and do not work with an already-connected client. But then passing a `pg.PoolClient` should be prevented by the function signature. 

Perhaps the `_zapatos` key should be required in `TxnClient` so I can't pass a regular `PoolClient` ([source](https://github.com/jawj/zapatos/blob/master/src/db/transaction.ts#L36)) ? Just guessing.

Thanks again

**Comment by tetchel** [comment]
ah, it looks like this is roughly the same issue as https://github.com/jawj/zapatos/issues/77

**Comment by tetchel** [comment]
@jawj sorry for the ping. I've been using a fork of this library for the last 6 months with this change implemented. I would really love to switch to the upstream. 

Would you accept a PR with this change?

**Comment by tetchel** [comment]
Wow, thank you! :grinning:

**Comment by jawj** [comment]
No problem, though a warning that this is currently not working as it should: I've not yet found a reliable way to actually distinguish a client from a pool. In my testing, `x instanceof pg.Pool` and `x instanceof pg.Client` seem both always to be false.

**Comment by jawj** [comment]
OK, this is now working and released in 5.0.3.
